Sage/carbon
# chore(eslint): update config so rules are applied to the correct files (FE-5758)

### Proposed behaviour

- Update eslint config by creating new overrides, so that:
   - Jest linting rules are only applied to `*.spec.*` files
   - Cypress linting rules are only applied to files within the `cypress/` directory
- Refactor any jest and cypress test code that threw lint errors due to these new config changes[^1]
- Remove `eslint-disable` comments in `ButtonToggle`'s cypress test

[^1]: Note these errors were not being flagged before since their corresponding rules were being overwritten by our config.

### Current behaviour

- Jest linting rules are being applied to cypress test files, which is incorrect[^2]
- Some recommended rules from both the jest and cypress eslint plugins are being overwritten, so these lint errors remain unflagged.

[^2]: Cypress uses Mocha internally as its test runner, so the cypress tests do not use jest in any way, which is why the jest lint rules should not be applied to cypress files

### Checklist




#### QA


### Testing instructions

Verify that:
1. Any eslint rules that would conflict with prettier are not loaded
2. The correct rules are being applied to jest test files, cypress test files, and other remaining files

A shell script has been created for convenience when testing these two criteria. Please see these notes on how to use it this script:
[Testing eslint config for FE-5758.pdf](https://github.com/Sage/carbon/files/11949567/Testing.eslint.config.for.FE-5758.pdf)
